src/Form/DomainMenuAccessSettingsForm.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Drupal\domain_menu_access\Form;
+
+use Drupal\Component\Plugin\Discovery\CachedDiscoveryInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class DomainMenuAccessSettingsForm extends ConfigFormBase {
+
+  /**
+   * Block manager
+   */
+  protected $blockManager;
+
+  /**
+   * Menu entities storage
+   */
+  protected $menu;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, CachedDiscoveryInterface $block_manager, EntityStorageInterface $menu_storage) {
```

Comment:
should be `\Drupal\Core\Block\BlockManagerInterface $block_manager, ConfigEntityStorageInterface $menu_storage`